redaxo/src/core/lib/packages/interface.php

```diff
@@ -152,6 +152,7 @@ public function isAvailable();
      * Returns if the package is installed.
      *
      * @return bool
+     * @psalm-assert-if-false false $this->isAvailable()
```

Comment:
Die psalm-Meldung, die unten angezeigt wird, kapiere ich nicht.

Aber unabhängig von der Meldung, gibt es folgendes Problem mit Methoden in den assertions:

> Please note that the example above only works if you enable [method call memoization](https://psalm.dev/docs/running_psalm/configuration/#memoizemethodcallresults) in the config file or annotate the class as [immutable](https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-immutable).

https://psalm.dev/docs/annotating_code/adding_assertions/#asserting-return-values-of-methods

Ich weiß nicht so recht, ob zu unserer Codebase "method call memoization" passt. Denke eher nicht.

Reply:
Ok schade. Diese einschrgibts bei phpstan nicht

Reply:
Ah, in phpstan scheint es die Einstellung [rememberPossiblyImpureFunctionValues](https://phpstan.org/config-reference#rememberpossiblyimpurefunctionvalues) zu sein, die dort aber default aktiv ist.

Reply:
Ich denke, wir sollen uns auf eine Variante festlegen, und die in beiden Tools dann verwenden.
Müssen mal schauen, wie viele false-positive das so sind.

Reply:
der change selbst ist nicht super wertvoll. wegen mir können wir auch erstmal hier schließen und die zeit woanders besser investieren

Reply:
ok dann schließen wir (erstmal)